bin/prettier.js

```diff
@@ -9,25 +9,8 @@ const minimist = require("minimist");
 const prettier = require("../index");
 
 const argv = minimist(process.argv.slice(2), {
-  boolean: [
-    "write",
-    "stdin",
-    "single-quote",
-    "trailing-comma",
-    "bracket-spacing",
-    // The supports-color package (a sub sub dependency) looks directly at
-    // `process.argv` for `--no-color` and such-like options. The reason it is
-    // listed here is to avoid "Ignored unknown option: --no-color" warnings.
-    // See https://github.com/chalk/supports-color/#info for more information.
-    "color",
-    "help",
-    "version",
-    "debug-print-doc",
-    "debug-check",
-    // Deprecated in 0.0.10
-    "flow-parser"
-  ],
-  string: ["print-width", "tab-width", "parser"],
+  boolean: true,
```

Comment:
I see. So the problem is that you want `--trailing-commas` to be either a boolean or a string (where the booleans are aliases for string values), but options must be one or the other, not both at the same time. Unless you pay the cost of `boolean: true`:

```
$ ./bin/prettier.js index.js --parser flow
Ignoring unknown --parser value, falling back to "babylon":
  Expected "flow" or "babylon", but received: ""
```

(Also note that the above is quite a breaking change.)

One solution is to always treat `--trailing-commas` as a string. If you want to turn on trailing commas you have to type either `--trailing-commas all` or `--trailing-commas compat`.